#!/usr/bin/env python
PACKAGE = "me_cs301_robots"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("leg1_j1",    double_t,    0, "leg1_j1 position", 0.0, -1.047, 2.0944)
gen.add("leg1_j2",    double_t,    0, "leg1_j2 position", 0.0, -1.5708, 3.14159)
gen.add("leg1_j3",    double_t,    0, "leg1_j3 position", 1.5708, 0.0, 3.14159)

gen.add("leg2_j1",    double_t,    0, "leg2_j1 position", 0.0,-1.047, 2.0944)
gen.add("leg2_j2",    double_t,    0, "leg2_j2 position", 0.0, -1.5708, 3.14159)
gen.add("leg2_j3",    double_t,    0, "leg2_j3 position", 1.5708, 0.0, 3.14159)

gen.add("leg3_j1",    double_t,    0, "leg3_j1 position", 0.0,-1.047, 2.0944)
gen.add("leg3_j2",    double_t,    0, "leg3_j2 position", 0.0, -1.5708, 3.14159)
gen.add("leg3_j3",    double_t,    0, "leg3_j3 position", 1.5708, 0.0, 3.14159)

gen.add("leg4_j1",    double_t,    0, "leg4_j1 position", 0.0,-1.047, 2.0944)
gen.add("leg4_j2",    double_t,    0, "leg4_j2 position", 0.0, -1.5708, 3.14159)
gen.add("leg4_j3",    double_t,    0, "leg4_j3 position", 1.5708, 0.0, 3.14159)

exit(gen.generate(PACKAGE, "me_cs301_robots", "RWJointControl"))
